gpage.h

```diff
@@ -222,16 +228,17 @@ namespace gcpp {
 	//
 	inline
 	bool gpage::contains(gsl::not_null<const byte*> p) const noexcept {
-		return &storage[0] <= p && p < &storage[total_size - 1];
+		auto cmp = std::less<>{};
+		return !cmp(p, &storage[0]) && cmp(p, &storage[total_size - 1]);
```

Comment:
@neilmacintosh (repeating something I've already said privately last week) Good point, please keep on keeping me honest. Even for those of us who value and were part of the designers/proposers of new basic types, especially `gsl::span`, it takes time to get in the habit of using them. I'm sure the same will happen as the world gradually gets used to using things like C++17 `string_view`...